plans/example-rust/src/main.rs

```diff
@@ -7,51 +8,43 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     let (client, _run_parameters) = testground::client::Client::new().await?;
     client.wait_network_initialized().await?;
 
-    let local_addr = &if_addrs::get_if_addrs()
-        .unwrap()
+    let local_addr = &if_addrs::get_if_addrs()?
         .into_iter()
         .find(|iface| iface.name == "eth1")
-        .unwrap()
+        .ok_or("Could not find interface eth1")?
         .addr
         .ip();
 
     match local_addr {
         std::net::IpAddr::V4(addr) if addr.octets()[3] == 2 => {
             println!("Test instance, listening for incoming connections.");
 
-            let listener = TcpListener::bind((*addr, LISTENING_PORT))?;
+            let listener = TcpListener::bind((*addr, LISTENING_PORT)).await?;
 
             client.signal("listening".to_string()).await?;
 
-            for _stream in listener.incoming() {
-                println!("Established inbound TCP connection.");
-                break;
-            }
+            println!("Established inbound TCP connection.");
+            let _ = listener.accept().await?;
```

Comment:
```suggestion
            let _ = listener.accept().await?;
            println!("Established inbound TCP connection.");
```
This is in the wrong order, right?

Reply:
yes, thanks!